src/settings-mgr.h

```diff
@@ -24,6 +29,13 @@ class SettingsManager : public QObject {
     unsigned int maxUploadRatio() { return maxUploadRatio_; }
     bool allowInvalidWorktree() { return allow_invalid_worktree_; }
     bool syncExtraTempFile() { return sync_extra_temp_file_; }
+    void getProxy(ProxyType &proxy_type, QString &proxy_host, int &proxy_port, QString &proxy_username, QString &proxy_password) {
+        proxy_type = use_proxy_type_;
```

Comment:
When the value of the param would be changed, We'd better pass it by a pointer instead of by reference. Although we don't have this rule explicitly in [our coding style guide](https://github.com/haiwen/seafile-client/tree/master/coding-style.md), we have followed it everywhere else of the project.